File: Afg1Stromrallye/Afg1Stromrallye.API/Board.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Afg1Stromrallye.API
{
    public class Board
    {
        public int?[,] Batteries { get; }

        public Vector2Int StartingPosition { get; }
        public int StartingBattery { get; }
    }
}

File: Afg1Stromrallye/Afg1Stromrallye.API.FSharp/Board.fs
namespace Afg1Stromrallye.API.FSharp

open System.Collections.Generic

module Board =

    type Board =
        val Width : int
        val Height : int
        val BatteryLocations : HashSet<Vector2Int>
        val Distances : Map<Vector2Int * Vector2Int, int>

    let build width height batteries =
        